source/manuals/programming-languages/html.html.md.erb

```diff
@@ -1,85 +1,153 @@
 ---
 title: HTML coding style
-last_reviewed_on: 2016-02-03
+last_reviewed_on: 2019-07-09
 review_in: 6 months
 ---
 
 # <%= current_page.data.title %>
 
-## HTML5
+## Browser support
 
-* Always quote attribute values - the decision tree for remember when to
-  selectively quote is too much mental overhead, so just quote them
-* Always use well-formed HTML - implicitly closed elements such as
-  `<p>` cause unnecessary parsing overhead and can cause issues in
-  accessibility
+The GOV.UK Service Manual lists the [minimum browsers that GDS is required
+to support](https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in).
+Tools such as GOV.UK Frontend, that are used to help build services, may have
+[stricter standards of browser support](https://github.com/alphagov/govuk-frontend#browser-support).
+Therefore, you will need to test your code in different browsers depending on
+which part of GDS you are working on.
 
-## Use [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) roles appropriately
+Whatever your level of browser support, HTML should be written in such a way
+that older browsers see an appropriate fallback or warning message. For example,
+some browsers do not support the `<video>` element, so you should ensure that
+some contextual information is included (the `<p>` element in this example):
 
-### Guidance on using [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) landmarks
+```html
+<video width="320" height="240" controls>
+  <source src="movie.mp4" type="video/mp4">
+  <source src="movie.ogg" type="video/ogg">
+  <p>Unable to load video. Please upgrade your browser.</p>
+</video>
+```
 
-When you use HTML5 sectioning elements it's a good idea to include
-the appropriate [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) landmark roles as well. The trick when you're
-using both HTML5 and [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) is to put the landmark on the sectioning
-element itself because this avoids the information being duplicated by
-screen readers that support both technologies.
+When using structual HTML5 elements, such as `<main>`, ensure that you include
+a [shim that allows them to be styled](https://github.com/aFarkas/html5shiv) in
+legacy browsers. You can include the script inside a `<!--[if lt IE 9]><![endif]-->`
```

Comment:
Are we sure that that the the html5 shim is currently used anywhere? I haven't come across it on the projects that I've worked on, so happy to be proved wrong. It also seems a little prescriptive - if we do need to include a section on shims, it feels like a line on why they might be useful could be better.

Reply:
The `aFarkas/html5shiv` isn't being used on GOV.UK, but we do have an internal one:

```html
<!--[if lt IE 9]><script src="https://assets.publishing.service.gov.uk/static/ie-a4524544a53d57a7e259b4bb966b9c32557c98c920b77e52d09304642b68401a.js" crossorigin="anonymous" integrity="sha256-pFJFRKU9V6fiWbS7lmucMlV8mMkgt35S0JMEZCtoQBo="></script><![endif]-->
```

This was in the original PR but was removed because [we don't want other teams hot-linking to our assets](https://github.com/alphagov/gds-way/pull/287/files/a3555155d27173262e110a19f86311c1194f853c#diff-870dd991fb2314bcc052a98935017b42) (I also don't know how that asset is generated or how long it's expected to exist - in my previous company, old assets would expire as new versions are released).